summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/VersionedBatchStore.scala

```diff
@@ -200,13 +201,19 @@ class VersionedBatchStore[K, V, K2, V2](rootPath: String, versionsToKeep: Int, o
       case _ => Some(batchVersion)
     }
 
-    lastVals.map(pack(batchID, _))
-      .toPipe((0, 1))
-      .write(VersionedKeyValSource[K2, V2](rootPath,
-        sourceVersion = None,
-        sinkVersion = newVersion,
-        maxFailures = 0,
-        versionsToKeep = versionsToKeep))
+    val target = VersionedKeyValSource[K2, V2](rootPath,
+      sourceVersion = None,
+      sinkVersion = newVersion,
+      maxFailures = 0,
+      versionsToKeep = versionsToKeep)
+    if (!target.sinkExists(mode)) {
+      logger.info(s"Versioned batched store version for $this @ $newVersion doesn't exist. Will write out.")
+      lastVals.map(pack(batchID, _))
+        .write(target)
+    } else {
+      logger.warn(s"""Versioned batched store version for $this @ $newVersion already exists! Will skip adding to plan.
+        | Note you must have at least one store that requires writing out or the job will fail.""".stripMargin('|'))
```

Comment:
this constraint is probably a pain point. So I think the real bug comes from crashed that happen after writing the store output, but before we commit to the WaitingState that we have processed a block of time.